flycheck.el

```diff
@@ -6385,7 +6385,9 @@ This will either be the directory that contains `mix.exs' or,
 if no such file is found in the directory hierarchy, the directory
 of the current file."
   (or
-   (locate-dominating-file (buffer-file-name) "mix.exs")
+   (and
+    buffer-file-name
+    (locate-dominating-file buffer-file-name "mix.exs"))
```

Comment:
I'm always using the function out of some habit that I can't explain :relaxed: but it's generally totally fine to use the variable instead.